xbmc/video/windows/GUIWindowVideoBase.cpp

```diff
@@ -1097,11 +1097,6 @@ bool CGUIWindowVideoBase::OnPlayMedia(int iItem, const std::string &player)
   CServiceBroker::GetPlaylistPlayer().SetCurrentPlaylist(PLAYLIST_NONE);
 
   CFileItem item(*pItem);
-  if (pItem->IsVideoDb())
-  {
-    item.SetPath(pItem->GetVideoInfoTag()->m_strFileNameAndPath);
-    item.SetProperty("original_listitem_url", pItem->GetPath());
-  }
```

Comment:
Are you sure removing those lines has no hidden side effect?
If there's code that just sets `m_strFileNameAndPath` in the `VideoInfoTag` playback will probably fail.

Reply:
Thanks for looking @a1rwulf. In fact it does...if I create a `.strm` file with a `videodb://` url that resolves to an iso file it will behave wrongly as before. Looks like we need to recursively resolve the urls (issue 1).

I also found regressions in the watched states due to the missing `"original_listitem_url"` (which sound like another legacy hack) (issue 2). It also looks like the widgets on the homemenu are populated with the `strFileNameAndPath` and not simply the path (issue 3).

So, I'll try to fix issue 1 and issue 3 for Leia. Issue 2 will keep the old code and I'll look into it when we branch. Marking this as WIP.

Reply:
mind taking another look @a1rwulf ?

Reply:
Any special reason why you moved this handling from `GUIWindowVideoBase.cpp` to `PlaylistPlayer`?
I know too little about the code, it looks like a hack anyway and I can't judge who (what class or subsystem) should be responsible to do this.

Reply:
They are different things. The block in `GUIWindowVideoBase.cpp` was removed and not moved. Kodi already sets the dynpath to the iso file at some point so it is wrong to change the original path.
The block in `PlaylistPlayer` is to allow the playback if the `videodb://` url is used inside a playlist (e.g. a .strm file).
I don't think we can call this PR an hack considering the state of `CFileItem` - it's the only fix we can have for leia.

Reply:
Despite the change from `Path` to `DynPath` the lines are the same thing.
I agree that changing the `Path` is wrong.
As you fixed my concern about breakage, I hope some other dev can judge if this behavior belongs to `PlayListPlayer`.